package/MDAnalysis/coordinates/MOL2.py

```diff
@@ -307,7 +307,12 @@ def encode_block(self, obj):
         obj : AtomGroup or Universe
         """
         # Issue 2717
-        obj = obj.atoms
+        try:
+            obj = obj.atoms
```

Comment:
I’d hope stuff is type checked by this point

Reply:
Unfortunately this is the failure point if you pass a Timestep object to the MOL2Writer. I could move it up to _write_next_frame if you think it's more appropriate?